### ISSUE TITLE
Bookmark bar stays visible after all bookmarks are deleted

### ISSUE THREAD

**Issue by btlechowski** [brave/browser-laptop#14047]
### Steps to Reproduce

1. Clean install
2. Click the star
3. Click 'Remove'

**Actual result:**
Bookmark bar stays visible
![image](https://user-images.githubusercontent.com/34715963/39724689-c45081fc-5249-11e8-9f40-41fb805bfff2.png)

**Expected result:**
Bookmark bar is hidden

**Reproduces how often:**
100%

### Brave Version
```
Brave: 0.22.702 
V8: 6.6.346.26 
rev: e4a853d9990c62328a8a4320fec21f58c9cdc0c1 
Muon: 6.0.7 
OS Release: 4.13.0-21-generic 
Update Channel: Beta 
OS Architecture: x64 
OS Platform: Linux 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 66.0.3359.139
```
Also reproduced on Mac and Windows by @srirambv @LaurenWags @kjozwiak 

**Reproducible on current live release:**
No. Tested v0.22.669

Cc @bsclifton @petemill

**Comment by petemill** [comment]
Note that this is also sort of reproducible in the current release, except that in the current release, the bookmarks toolbar gets really short, which itself was considered a bug since a user could not have the bookmarks toolbar with normal height with no bookmark items in order to interact with it.
However, I'd be happy to look at fixing for this release, since it's now more obvious with the fix from #13758

**Comment by btlechowski** [comment]
Verified on Ubuntu 17.10 x64
* 0.22.703 903b8d0d
* libchromiumcontent 66.0.3359.139
* muon: 6.0.8
